src/v/pandaproxy/schema_registry/storage.h

```diff
@@ -1492,8 +1492,18 @@ struct consume_to_store {
               offset);
             if (!val) {
                 try {
+                    auto schema = co_await _store.get_subject_schema(
+                      key.sub, key.version, include_deleted::yes);
+
                     co_await _store.delete_subject_version(
                       key.sub, key.version, force::yes);
+
+                    auto remaining_subjects
+                      = co_await _store.get_schema_subjects(
+                        schema.id, include_deleted::yes);
+                    if (remaining_subjects.empty()) {
+                        co_await _store.delete_schema(schema.id);
+                    }
```

Comment:
LGTM, but storage is the wrong place for this kind of logic, it should go into `sharded_store::delete_subject_version`.

Reply:
Done